saleor/order/views.py

```diff
@@ -31,12 +31,9 @@ def details(request, token):
     orders = orders.select_related(
         'billing_address', 'shipping_address', 'user')
     order = get_object_or_404(orders, token=token)
-    notes = order.notes.filter(is_public=True)
-    ctx = {'order': order, 'notes': notes}
-    if order.is_open():
-        user = request.user if request.user.is_authenticated else None
-        note = OrderNote(order=order, user=user)
-        note_form = OrderNoteForm(request.POST or None, instance=note)
+    ctx = {'order': order}
+    if order.is_open() and not order.customer_note:
+        note_form = CustomerNoteForm(request.POST or None, instance=order)
         ctx.update({'note_form': note_form})
```

Comment:
What about we set `note_form` to `None` and then, we move `ctx = {...}` at the end and add everything we need to the context to save some useless instructions?

```python
def details(request, token):
    note_form = None
    orders = Order.objects.confirmed().prefetch_related(
        'lines__variant', 'fulfillments', 'fulfillments__lines',
        'fulfillments__lines__order_line')
    orders = orders.select_related(
        'billing_address', 'shipping_address', 'user')
    order = get_object_or_404(orders, token=token)
    if order.is_open() and not order.customer_note:
        note_form = CustomerNoteForm(request.POST or None, instance=order)
        ctx.update({'note_form': note_form})
        if request.method == 'POST':
            if note_form.is_valid():
                note_form.save()
                return redirect('order:details', token=order.token)
    fulfillments = order.fulfillments.filter(
        status=FulfillmentStatus.FULFILLED)
    ctx = {
        'order': order, 'fulfillments': fulfillments, 'note_form': note_form}
    return TemplateResponse(request, 'order/details.html', ctx)